pages/api/skillRatingsFunctions.ts

```diff
@@ -0,0 +1,56 @@
+import { SkillsAndRatings } from "../../graphql/fetchSkillsAndRatings";
+import { User } from "../../graphql/fetchUserProfile";
+import { UserSkillsRatings } from "../../graphql/fetchUserSkillsRatings";
+import { SkillRatingsRow } from "../../redux/skillRatingsSlice";
+
+  export const transformSkillRating = (skillRatings: UserSkillsRatings[]) => {
+    // map to redux type for skillRatings page
+    const mappedSkillRating: SkillRatingsRow[] = skillRatings.map((row) => {
+      return {
+        userSkillId: row.id,
+        skillId: row.intro_course_skill["id"],
+        skillName: row.intro_course_skill["name"],
+        unitName: row.intro_course_skill["intro_course_unit"]["title"],
+        studentRating: parseInt(row.studentRating),
+      };
+    });
+
+    return mappedSkillRating;
+  };
+
+  export const transformSkillsAndRatings = (skillRatings: SkillsAndRatings[]) => {
+     // map to redux type for profileV2 page
+    return skillRatings.map((row) => {
+      return {
+        skillId: row.id,
+        skillName: row.name,
+        unitName: row.intro_course_unit.title,
+        studentRating: row.intro_course_skills_users_aggregate.nodes.length > 0 
+        ? parseInt(row.intro_course_skills_users_aggregate.nodes[0].studentRating)
+        : 0,
+        userSkillId: row.intro_course_skills_users[0]?.id ? row.intro_course_skills_users[0].id : Math.random(),
+      };
+    });
+  };
+
+  export const transformSkillRatingForDB = (skillRatings: SkillRatingsRow[], user:User) => {
+    // map from redux type to write back to DB
+    const transformedOutput = skillRatings.map((row) => {
+      if (typeof row.userSkillId == "number") {
```

Comment:
I think we can ignore this case. User IDs should probably always be strings.

If they are numbers in the database that is because they created their user account before I migrated our auth system. All new users will have strings

Reply:
see comment above.